scripts/release_notes.rb

```diff
@@ -7,57 +7,91 @@
 require 'uri'
 require 'net/http'
 require 'json'
-require 'pry-byebug'
 
 URL = 'https://projects.theforeman.org'
 
-# Check out installer modules
-# Compare packages in nightly vs release and retag dependencies
-
-# Create new release via API call
-# Create new search by open + release field
-
-unless @current_release_id = ARGV[0]
-  puts "Usage: #{$0} RELEASE_ID"
-  puts "Hint: 1.16.0 is 240"
+unless ARGV.size == 2
+  puts "Usage: #{$0} PROJECTS RELEASE_NAME"
+  puts "Example: #{$0} foreman,puppet-foreman,selinux,smart-proxy 1.18.0"
   exit 1
 end
 
-# Move bugs with Release set to VERSION to NEXT-VERSION with status new
-def gather_issues(offset=0)
-  url = "#{URL}/projects/foreman/issues.json?status_id=closed&offset=#{offset}&limit=100&release_id=#{@current_release_id}"
+@project_names = ARGV[0].split(',')
+@current_release_name = ARGV[1]
+
+def url_to_json(url)
   uri = URI(URI.escape(url))
   response = Net::HTTP.get(uri)
-	result = JSON.parse(response)
-	if result['total_count'].to_i - offset - 100 <= 0
-		result['issues']
-	else
-		result['issues'] += gather_issues(offset + 100)
-	end
+  JSON.parse(response)
+end
+
+# Determine version
+def get_version_id(project)
+  url = "#{URL}/projects/#{project}/versions.json"
+  result = url_to_json(url)
+  return nil if result['versions'].nil?
+  result = result['versions'].detect{|v| v['name'] == @current_release_name }
+  return nil if result.nil?
+  result['id']
+rescue
+  puts "Error getting version list from #{url}"
+  exit 1
+end
+
+@current_release_ids = @project_names.map{|project| get_version_id(project)}.compact
+
+if @current_release_ids.empty?
+  puts "Release #{@current_release_name} not found in any of the projects #{@project_names}"
+  exit 1
+end
+
+def releases_filter
+  @current_release_ids.map{|release| "v[cf_12][]=#{release}"}.join('&')
 end
 
-def modify_target_version!(issue_id, options)
-  uri = URI(URI.escape("#{URL}/issues/#{issue_id}.json"))
-  req = Net::HTTP::Put.new(
-		uri,
-		{ 'Content-Type' => 'application/json',
-		  'X-Redmine-API-Key' => File.read('~/.redminekey') }
-	)
-  Net::HTTP.new(uri.host, uri.port).start {|http| http.request(req) }
+def gather_issues(offset = 0, feature = true)
```

Comment:
I wonder if we should split this by `tracker` in the result rather than retrieve it twice.